src/MIDebugEngine/Engine.Impl/DebugUnixChildProcess.cs

```diff
@@ -0,0 +1,260 @@
+                    _state = State.AtExec;
+                    await _process.MICommandFactory.ExecContinue();  // run the child
+                }
+            }
+        }
+
+        private async Task ContinueTheChild()
+        {
+            Debug.Assert(_state == State.AtExec, "wrong vfork processing state");
+            if (_newpid != 0)
+            {
+                int inf = _process.InferiorByPid(_newpid);
+                if (inf != 0)
+                {
+                    await _process.ConsoleCmdAsync("inferior " + inf.ToString());
+                    await _process.MICommandFactory.ExecContinue();  // run the child
+                }
+            }
+        }
+
+        private async Task<bool> DetachAndContinue()
+        {
+            int inf = _process.InferiorByPid(_newpid);
+            if (inf == 0)
+                return false;    // cannot process the child
+            await _process.ConsoleCmdAsync("inferior " + inf.ToString());
+            await _process.MICommandFactory.TargetDetach();     // detach from the child
+            await _process.ConsoleCmdAsync("inferior 1");
+            lock(AD7Engine.ChildProcessLaunch)
+            {
+                AD7Engine.ChildProcessLaunch.Add(_newpid);
+            }
+            string engineName;
+            Guid engineGuid;
+            _process.Engine.GetEngineInfo(out engineName, out engineGuid);
+            _launchOptions.BaseOptions.ProcessId = _newpid;
+            _launchOptions.BaseOptions.ProcessIdSpecified = true;
+            //_process.Callbacks.OnDebugLaunchedProcessAttachRequest(_newpid, _process.Engine);
+            _state = State.Enabled;
+            HostDebugger.Attach(_launchOptions.ExePath, LaunchOptions.GetOptionsString(_launchOptions.BaseOptions), engineGuid);
+            await _process.MICommandFactory.ExecContinue();     // continue the parent
+            return true;   // parent is running
+        }
+
+        public async Task Enable(bool enable)
+        {
+            if (enable && _state == State.Init)
+            {
+                await _process.MICommandFactory.Set("detach-on-fork", "off");
+                await _process.MICommandFactory.Catch("fork");
+                await _process.MICommandFactory.Catch("vfork");
+                _state = State.Enabled;
+                _newpid = 0;
+            }
+            else if (!enable)
+            {
+                if (_forkBp != null)
+                {
+                    await _process.MICommandFactory.BreakDelete(_forkBp);
+                    _forkBp = null;
+                }
+                if (_vforkBp != null)
+                {
+                    await _process.MICommandFactory.BreakDelete(_vforkBp);
+                    _vforkBp = null;
+                }
+                _state = State.Init;
+            }
+        }
+
+        public async Task SetBreakAtMain()
+        {
+            Results results = await _process.MICommandFactory.BreakInsert("main", null);
```

Comment:
Yeah, this could be an issue. I cannot find any way to discover the entry point name for a process. The gdb command 'start' seems to be independent of the name of the entry point, but it cannot be used in this ('exec') context.